### PR TITLE
Adding update_sync_token functionality to Appconfig Clients

### PR DESCRIPTION
**Pull request by seankane-msft**
Breaking #16771  into two PR's. This one handles the `update_sync_token` functionality.

### CODE REVIEW

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py [hunk 1]
```diff
@@ -578,3 +581,19 @@ def set_read_only(
...
+
+    def update_sync_token(self, token):
+        # type: (str) -> None
+
+        """Add a sync token to the internal list of tokens.
+        :param token: The sync token to be added to the internal list of tokens
+        :type token: str
+        """
+
+        self._sync_token_policy.add_token(token)
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:593]: Given you have self._sync_token_policy = None above, do we need to check if self._sync_token_policy before calling add?

File: sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py [hunk 2]
```diff
@@ -464,3 +465,40 @@ def test_set_read_only(self, client, appconfiguration_connection_string, test_co
         with pytest.raises(ResourceModifiedError):
             client.set_read_only(set_kv, True, match_condition=MatchConditions.IfNotModified)
 
+    @app_config_decorator
+    def test_sync_tokens(self, client):
+
+        sync_tokens = copy.deepcopy(client._sync_token_policy._sync_tokens)
+        keys = list(sync_tokens.keys())
+        seq_num = sync_tokens[keys[0]].sequence_number
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client.py:473]: I am afraid it is guaranteed there is only one key in sync_tokens.keys() and the updated one must be sync_tokens[keys[0]].

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py (hunk 1 above)
Comment (seankane-msft) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py:593]: Yes, good catch. If the user provides a `pipeline` kwarg when instantiating the client, they are giving up the ability to use sync tokens.

File: sdk/appconfiguration/azure-appconfiguration/CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,9 @@
 
 ## 1.1.2 (Unreleased)
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration/CHANGELOG.md:6]: We are adding a new feature, need 1.2.0 :)

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**